src/silo/preprocessing/metadata_info.cpp

```diff
@@ -0,0 +1,205 @@
+      [](const duckdb::Value& value) { return value.GetValue<std::string>(); }
+   );
+   return return_value;
+}
+
+std::unordered_map<std::string, std::string> validateFieldsAgainstConfig(
+   const std::unordered_map<std::string, std::string> found_metadata_fields,
+   const silo::config::DatabaseConfig& database_config
+) {
+   std::vector<std::string> config_metadata_fields;
+   std::transform(
+      database_config.schema.metadata.begin(),
+      database_config.schema.metadata.end(),
+      std::back_inserter(config_metadata_fields),
+      [](auto metadata) { return metadata.name; }
+   );
+
+   std::unordered_map<std::string, std::string> validated_metadata_fields;
+   for (const auto& pair : found_metadata_fields) {
+      if (std::find(config_metadata_fields.begin(), config_metadata_fields.end(), pair.first)
+       != config_metadata_fields.end()) {
+         validated_metadata_fields.emplace(pair);
+      }
+   }
+   for (const std::string& name : config_metadata_fields) {
+      if (!validated_metadata_fields.contains(name)) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "The metadata field {} which is contained in the database config is "
+            "not contained in the input.",
+            name
+         ));
+      }
+   }
+
+   std::string metadata_field_string;
+   for (const auto& [field, select] : validated_metadata_fields) {
+      metadata_field_string += "'" + field + "' with selection '" + select + "',";
+   }
+   SPDLOG_TRACE("Found metadata fields: " + metadata_field_string);
+   if (!validated_metadata_fields.contains(database_config.schema.primary_key)) {
+      throw silo::preprocessing::PreprocessingException(
+         "Primary key not present in both database config and metadata file."
+      );
+   }
+   return validated_metadata_fields;
+}
+
+void detectInsertionLists(
+   const std::filesystem::path& ndjson_file,
+   std::unordered_map<std::string, std::string>& metadata_fields_to_validate
+) {
+   duckdb::DuckDB db(nullptr);
+   duckdb::Connection connection(db);
+   auto top_level_entries =
+      connection.Query(fmt::format("SELECT * FROM '{}' LIMIT 0", ndjson_file.string()));
+   for (size_t idx = 0; idx < top_level_entries->ColumnCount(); idx++) {
+      const std::string& top_level_entry = top_level_entries->ColumnName(idx);
+      if (top_level_entry == "nucleotideInsertions" || top_level_entry == "aminoAcidInsertions") {
+         auto contained_insertions = connection.Query(
+            fmt::format("SELECT {}.* FROM '{}' LIMIT 0", top_level_entry, ndjson_file.string())
+         );
+         if (contained_insertions->ColumnCount() == 0) {
+            metadata_fields_to_validate[top_level_entry] = "''";
+         }
+         if (contained_insertions->ColumnCount() == 1) {
+            metadata_fields_to_validate[top_level_entry] = fmt::format(
+               "list_string_agg({}.{})", top_level_entry, contained_insertions->ColumnName(0)
```

Comment:
Just for my understanding: Is this the same as
```
               "list_string_agg({0}.{1})", top_level_entry, contained_insertions->ColumnName(0)
```
?

Reply:
Yes